File: dailyChallenges/0021-IsSubsequence/21-IsSubsequence.cpp
class Solution {
public:
    bool isSubsequence(string s, string t) {
        int i = 0;
        int j = 0;

        if(s.length() > t.length())
        {
            return false;
        }

        while ((i < s.length()) && (j < t.length()))
        {
            if(s[i] == t[j])
            {
                i ++;
            }
            j ++;
        }

        return i == s.length();

    }
};

File: dailyChallenges/0021-IsSubsequence/21-IsSubsequence.go
func isSubsequence(s string, t string) bool {

	i, j := 0, 0

	if len(s) > len(t) {
		return false
	}

	for i < len(s) {
		for j < len(t) && s[i] != t[j] {
			j++
		}
		if j == len(t) {
			return false
		}
		i++
		j++
	}
	return true
}